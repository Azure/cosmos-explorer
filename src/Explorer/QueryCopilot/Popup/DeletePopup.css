.popupRoot {
    box-sizing: border-box;
    flex-direction: column;
    display: flex;
    align-items: flex-start;
    justify-content: center;
    padding: 16px 24px;
    gap: 16px;
    position: fixed;
    background: #FFFFFF;
    max-width: 880px;
    min-width: 280px;
    width: 70%;
    height: auto;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    border: 1px solid rgba(204, 204, 204, 0.5);
    box-shadow: 0px 6.4px 14.4px rgba(0, 0, 0, 0.132), 0px 1.2px 3.6px rgba(0, 0, 0, 0.108);
  }

.deleteCodeHeader {
    max-width: 100%;
    width: calc(100% - 48px);
    height: 24px;
    font-family: 'Segoe UI';
    font-style: normal;
    font-weight: 600;
    font-size: 18px;
    line-height: 24px;
    color: #292827;
    flex: none;
    order: 0;
    align-self: stretch;
    flex-grow: 0;
}

.deleteCodeText {
    max-width: 100%;
    width: calc(100% - 48px);
    height: 18px;
    font-family: 'Segoe UI';
    font-style: normal;
    font-weight: 400;
    font-size: 13px;
    line-height: 18px;
    color: #292827;
    flex: none;
    order: 1;
    align-self: stretch;
    flex-grow: 0;
}

.popUpButtonsBar {
    display: inline-block;
    margin: 0px 10px;
    margin-top: 60px;
    width: 100%;
}
.popUpButtonsBar > * + * {
    margin-left: 10px;
  }

.deletePopUpButton {
    box-sizing: border-box;
    flex-direction: row;
    align-items: center;
    padding: 10px 20px;
    gap: 10px;
    width: fit-content;
    height: 24px;
    background: #0078D4;
    border-radius: 2px;
}
.deletePopUpButton:focus {
    outline: none;
    box-shadow: none;
  }

.cancelPopupButton {
    box-sizing: border-box;
    flex-direction: row;
    align-items: center;
    padding: 10px 20px;
    gap: 10px;
    width: fit-content;
    height: 24px;
    background: #FFFFFF;
    border: 1px solid #8A8886;
    border-radius: 2px;
}
.cancelPopupButton:focus {
    outline: none;
    box-shadow: none;
  }

.commandPopupButton {
    background-color: #FFF8F0
}