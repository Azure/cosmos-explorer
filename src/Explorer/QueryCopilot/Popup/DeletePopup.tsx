import { DefaultButton, Modal, PrimaryButton, Stack, Text } from "@fluentui/react";
import React, { Dispatch, SetStateAction } from "react";

export const DeletePopup = ({
  showDeletePopup,
  setShowDeletePopup,
  setQuery,
  clearFeedback,
  showFeedbackBar,
}: {
  showDeletePopup: boolean;
  setShowDeletePopup: Dispatch<SetStateAction<boolean>>;
  setQuery: Dispatch<SetStateAction<string>>;
  clearFeedback: Dispatch<SetStateAction<void>>;
  showFeedbackBar: Dispatch<SetStateAction<boolean>>;
}): JSX.Element => {
  const deleteCode = () => {
    setQuery("");
    setShowDeletePopup(false);
    clearFeedback();
    showFeedbackBar(false);
  };

  return (
    <Modal
      isOpen={showDeletePopup}
      styles={{ main: { minHeight: "122px", minWidth: "880px" } }}
      titleAriaId="deleteDialogTitle"
      subtitleAriaId="deleteDialogSubTitle"
    >
      <Stack style={{ padding: "16px 24px", height: "auto" }}>
        <Text id="deleteDialogTitle" style={{ height: 24, fontSize: "18px" }}>
          <b>Delete code?</b>
        </Text>
        <Text id="deleteDialogSubTitle" style={{ marginTop: 10, marginBottom: 20 }}>
          This will clear the query from the query builder pane along with all comments and also reset the prompt pane
        </Text>
        <Stack horizontal tokens={{ childrenGap: 10 }} horizontalAlign="start">
          <PrimaryButton style={{ padding: "0px 20px", height: 24 }} onClick={deleteCode}>
            Delete
          </PrimaryButton>
          <DefaultButton style={{ padding: "0px 20px", height: 24 }} onClick={() => setShowDeletePopup(false)}>
            Close
          </DefaultButton>
        </Stack>
      </Stack>
    </Modal>
  );
};
