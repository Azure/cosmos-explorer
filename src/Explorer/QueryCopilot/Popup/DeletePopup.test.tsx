import { mount, shallow } from "enzyme";
import React from "react";
import { DeletePopup } from "./DeletePopup";

describe("Delete Popup snapshot test", () => {
  const setShowDeletePopupMock = jest.fn();
  const setQueryMock = jest.fn();
  const clearFeedbackMock = jest.fn();
  const showFeedbackBarMock = jest.fn();

  it("should render when showDeletePopup is true", () => {
    const wrapper = shallow(
      <DeletePopup
        showDeletePopup={true}
        setShowDeletePopup={setShowDeletePopupMock}
        setQuery={setQueryMock}
        clearFeedback={clearFeedbackMock}
        showFeedbackBar={showFeedbackBarMock}
      />
    );
    expect(wrapper.find("Modal").prop("isOpen")).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should not render when showDeletePopup is false", () => {
    const wrapper = shallow(
      <DeletePopup
        showDeletePopup={false}
        setShowDeletePopup={setShowDeletePopupMock}
        setQuery={setQueryMock}
        clearFeedback={clearFeedbackMock}
        showFeedbackBar={showFeedbackBarMock}
      />
    );
    expect(wrapper.props().children.props.showDeletePopup).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should call setQuery with an empty string and setShowDeletePopup(false) when delete button is clicked", () => {
    const wrapper = mount(
      <DeletePopup
        showDeletePopup={true}
        setShowDeletePopup={setShowDeletePopupMock}
        setQuery={setQueryMock}
        clearFeedback={clearFeedbackMock}
        showFeedbackBar={showFeedbackBarMock}
      />
    );

    wrapper.find("PrimaryButton").simulate("click");

    expect(setQueryMock).toHaveBeenCalledWith("");
    expect(setShowDeletePopupMock).toHaveBeenCalledWith(false);
  });

  it("should call setShowDeletePopup(false) when close button is clicked", () => {
    const setShowDeletePopupMock = jest.fn();
    const wrapper = mount(
      <DeletePopup
        showDeletePopup={true}
        setShowDeletePopup={setShowDeletePopupMock}
        setQuery={setQueryMock}
        clearFeedback={clearFeedbackMock}
        showFeedbackBar={showFeedbackBarMock}
      />
    );

    wrapper.find("DefaultButton").at(1).simulate("click");

    expect(setShowDeletePopupMock).toHaveBeenCalledWith(false);
  });

  it("should render the appropriate text content", () => {
    const wrapper = shallow(
      <DeletePopup
        showDeletePopup={true}
        setShowDeletePopup={setShowDeletePopupMock}
        setQuery={setQueryMock}
        clearFeedback={clearFeedbackMock}
        showFeedbackBar={showFeedbackBarMock}
      />
    );

    const textContent = wrapper.find("Text").map((text) => text.props().children);

    expect(textContent).toEqual([
      <b>Delete code?</b>,
      "This will clear the query from the query builder pane along with all comments and also reset the prompt pane",
    ]);
  });

  it("should have the correct inline style", () => {
    const wrapper = shallow(
      <DeletePopup
        showDeletePopup={true}
        setShowDeletePopup={setShowDeletePopupMock}
        setQuery={setQueryMock}
        clearFeedback={clearFeedbackMock}
        showFeedbackBar={showFeedbackBarMock}
      />
    );

    const stackStyle = wrapper.find("Stack[style]").props().style;

    expect(stackStyle).toEqual({ padding: "16px 24px", height: "auto" });
  });
});
