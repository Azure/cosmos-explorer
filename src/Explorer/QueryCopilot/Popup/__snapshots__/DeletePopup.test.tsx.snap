// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Popup snapshot test should not render when showDeletePopup is false 1`] = `
<Modal
  isOpen={false}
  styles={
    {
      "main": {
        "minHeight": "122px",
        "minWidth": "880px",
      },
    }
  }
  subtitleAriaId="deleteDialogSubTitle"
  titleAriaId="deleteDialogTitle"
>
  <Stack
    style={
      {
        "height": "auto",
        "padding": "16px 24px",
      }
    }
  >
    <Text
      id="deleteDialogTitle"
      style={
        {
          "fontSize": "18px",
          "height": 24,
        }
      }
    >
      <b>
        Delete code?
      </b>
    </Text>
    <Text
      id="deleteDialogSubTitle"
      style={
        {
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
    >
      This will clear the query from the query builder pane along with all comments and also reset the prompt pane
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="start"
      tokens={
        {
          "childrenGap": 10,
        }
      }
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        style={
          {
            "height": 24,
            "padding": "0px 20px",
          }
        }
      >
        Delete
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          {
            "height": 24,
            "padding": "0px 20px",
          }
        }
      >
        Close
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Delete Popup snapshot test should render when showDeletePopup is true 1`] = `
<Modal
  isOpen={true}
  styles={
    {
      "main": {
        "minHeight": "122px",
        "minWidth": "880px",
      },
    }
  }
  subtitleAriaId="deleteDialogSubTitle"
  titleAriaId="deleteDialogTitle"
>
  <Stack
    style={
      {
        "height": "auto",
        "padding": "16px 24px",
      }
    }
  >
    <Text
      id="deleteDialogTitle"
      style={
        {
          "fontSize": "18px",
          "height": 24,
        }
      }
    >
      <b>
        Delete code?
      </b>
    </Text>
    <Text
      id="deleteDialogSubTitle"
      style={
        {
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
    >
      This will clear the query from the query builder pane along with all comments and also reset the prompt pane
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="start"
      tokens={
        {
          "childrenGap": 10,
        }
      }
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        style={
          {
            "height": 24,
            "padding": "0px 20px",
          }
        }
      >
        Delete
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
        style={
          {
            "height": 24,
            "padding": "0px 20px",
          }
        }
      >
        Close
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;
